### PR TITLE
Add support for `--debug`

### LINKED ISSUE #322: Debug mode
I sometimes have issues with eslint_d. To understand what is happening, I'm adding some log instructions to the code. It would be nice if eslint_d had a debug mode builtin. This can be done by writing log to stderr if that doesn't conflict with the client/daemon communication. Otherwise, this can be done by writing log to a temporary file.

### PR DESCRIPTION
**Pull request by mantoni**
This uses the `debug` module that ships with `eslint` to print debug output to a separate terminal.

Usage notes are in the second commit.

Fixes #322

### CODE REVIEW

**Comment by mantoni** [comment]
I'm pretty happy with this now. You can have a server instance with debug logging attached to a new terminal like this:

```bash
eslint_d restart --debug # log everything
DEBUG="eslint_d:*" eslint_d restart --debug # log eslint_d only
```

You can also get debug output per invocation without restarting the server:

```bash
eslint_d test.js --debug
```

And it's possible to combine both, server restart with `--debug` and client calls with `--debug` 🤓

What do you think @DamienCassou and @L2jLiga?

**Comment by L2jLiga** [comment]
Hey, currently I have no much time, gonna check it this weekend

**Comment by DamienCassou** [comment]
This is really good, thank you @mantoni.

File: README.md [hunk 1]
```diff
@@ -174,6 +174,26 @@ The server is also automatically restarted if one of the following files
...
+`eslint_d` only:
+
+```bash
+❯ DEBUG="eslint_d:*"
+```
+
+For server side debug output, restart with `--debug` in a separate terminal:
+
+```bash
+❯ eslint_d restart --debug
```
Comment (DamienCassou) [inline README.md:192]: maybe we could also have

```session
$ DEBUG="eslint_d:*" eslint_d restart --debug
```

as example to only see "eslint_d" log?

**Review by DamienCassou — APPROVED** [review]

**Comment by mantoni** [comment]
📦 Released in `v14.2.0`

**Comment by L2jLiga** [comment]
I'm a bit late, but looks awesome!

**Pull request merged**